Add httpwg chair-tooling skill (agenda subcommand) (#139)
httpwg/wg-materials: .claude/skills/httpwg/SKILL.md

or someone they've delegated to. There's no enforcement, and none is needed:
the consequential steps are gated by access anyway. You can't push to
`gh-pages` without commit rights, and you can't read presenter confirmations
or draft the mail without the chair's mailbox; a non-chair gets at most a
local, unpublishable draft.

## Prerequisites — state these plainly if missing

This skill leans on two capabilities. If either is unavailable, say so
before starting rather than silently working around it:

- **The ietf-llm MCP tool** — the queryable record of the httpbis group
  (charter, drafts, RFCs, minutes, mailing list, and meeting logistics).
  It is the primary source of truth for "what's the group doing" and,
  increasingly, for the mechanical facts below. Run `list_corpora` first;
  if `httpbis` is stale by more than a few days, run
  `start_gather(corpus="httpbis")` before relying on its activity data.
- **Access to the chair's email** — to find presenter confirmations, look
  up author addresses, and compose the call-for-presenters draft (Step 7).
  Any tool that can search the chair's mailboxes and create a draft will

>   do; this skill doesn't depend on a specific mail integration.

Author: Don't rifle through the chair's e-mails unbidden.

Author: Fixed in 0de7201 — the prerequisite plus a new up-front principle: touch the chair's mail only at their direction, never unbidden.

second-hand feed (a calendar entry, a stale cache) for any of them.

### Step 1 — pin down the meeting logistics

You need: the meeting number, the date, the **meeting-local** start/end
time, the room, and the session id (which the Meetecho URLs are built from).

Call `meeting_sessions(corpus="httpbis", meeting="<NNN>")`. It returns
every session live from datatracker with the venue-**local** weekday/date
and start–end time (DST-correct), the room, the session id, and both
Meetecho URLs (remote + onsite) already built — so you don't assemble or
convert anything by hand. From its output, settle:

- **How many sessions** the group has. The last couple of meetings have
  been single-session, but httpbis has often had **two**, so don't assume —
  check. Each session is a separate dated block with its own date, time,
  room, and session id; note each one's duration, since the time budget is
  **per session**, not pooled (see Step 4 and the multi-session layout
  below).
- **Use the local times verbatim.** The tool has already converted from
  UTC; never quote a UTC start (or a datatracker row-anchor token like
  `0700`) as if it were the local time.

> ### Step 2 — create the file in house format

Author: ```suggestion
### Step 2 — create a template agenda to fill in
```

Author: Applied in 0de7201.

### Active Drafts

_See also the [extensions listing](https://httpwg.org/http-extensions/)_

<one bullet per active WG draft>

### Other Topics

<one bullet per individual/other item that has requested time>
```

Canonical bullet format:

```
* <N> min - [<Title>](<datatracker-url>) - <Presenter>
```

- The presenter name comes only once they've confirmed (Step 5); a bare
  draft line with no name is correct until then.
- Append `(remote)` after the name when they're dialling in.
- Append `([slides](<file>.pdf))` once slides are uploaded — they live in
  the same `ietf<NNN>/` directory as the agenda.

The "Session I/II/III" label is the timeslot ordinal on that day (Friday
09:00–11:00 is Session I; a late slot might be Session IV).

The section set and their order aren't fixed. Beyond **Active Drafts** and
**Other Topics**, recent agendas also use **Related Work** (relevant
non-httpbis drafts) and the occasional one-off like **AD-Requested
Feedback** (a draft an AD has asked the WG to look at) — and they sometimes
lead with Other Topics rather than Active Drafts. Take the actual sections,
their order, and any extra Administrivia line (e.g. "10 min - Rechartering")

> from the most recent agenda, not the skeleton above.

Author: This will perpetuate one-off adjustments -- don't be so literal about copying things from the most recent agenda. 

What's more important is that you discuss with the chair what the requirements for THIS session are.

Author: Fixed in 0de7201 — Step 2 now treats recent agendas as a guide to formatting only; sections and items are this meeting's, worked out with the chair, and a past agenda's one-offs aren't carried forward.

be days apart (e.g. a Monday and a Thursday), so a presenter's stated
availability can decide placement.

### Step 3 — decide which drafts are in play, and verify status

The agenda should carry **active WG drafts that still need working-group
time**, plus individual/other drafts only where someone has asked to
present.

Get the candidate set from ietf-llm (`overview` / `read_digest`), then run
`draft_status(name)` on each candidate and let its **agenda-eligibility**
signal decide:

- **in the WG** (`I-D Exists`) → eligible for the agenda.
- **past the WG** (IESG processing) or **published** → the WG's work is
  essentially done; leave it off unless an author specifically wants a
  status slot.
- **dead** (expired / replaced) → off.

`draft_status` reads live datatracker state, so it catches what `overview`'s
cached active-draft list misses — that list can omit an active draft or keep
one that has since advanced, so treat it as a starting point and let
`draft_status` be the authority.

Individual drafts (e.g. `draft-<author>-httpbis-*`) belong under **Other
Topics**, and only when the author has requested time — see Step 5.

> 

Author: The idea here is that you're creating a candidate list that the chair can adjust.

To that end, ASK the chair if they want anything added/skipped. They might ask you to look up an e-mail thread with a discussion about an extra presentation, for example.

Author: Fixed in 0de7201 — Step 3 now frames the draft list as a candidate to put to the chair (add/skip), and says to have them point you at a relevant thread rather than going to look.

- **New work / other items → 10 or 15 minutes** by complexity; rarely more
  unless negotiated with the authors.
- **Negotiated allocations stand.** If a presenter agreed a length over
  mail, keep it.

To gauge activity, lean on two ietf-llm signals:

- **Open issues per draft** — from the gathered GitHub issues
  (`read_digest(corpus="httpbis", kind="issues", ...)` / `search_corpus`).
- **Recent list traffic** — `read_digest(corpus="httpbis", kind="threads",
  sort="activity", since=...)` / `search_corpus`.

If `list_corpora` shows httpbis without an `issues` source, the issue signal
is just missing — that's a gather-configuration gap (re-gather httpbis with
issues enabled), not a reason to hit the GitHub API directly. Note the gap
and size slots on list traffic and your own read of the drafts.

For each session, add up Administrivia + its items and confirm the total ≤
that session's slot length. State each session's total and headroom when
you present the agenda.

### Step 5 — presenters and confirmations

Only put a **presenter's name** on an item once they've **confirmed** —
an unconfirmed draft gets no name. Search the chair's mail for requests

> like "I'd like to present X at IETF <NNN>"; honour any length or

Author: ask the chair if they have any e-mails, don't just go looking.

Author: Fixed in 0de7201 — Step 5 now asks the chair about confirmations instead of searching their mail.

ordering agreed there, and note items explicitly moved elsewhere (e.g. an
author moving a draft to another WG — drop it).

For drafts that need time but have **no confirmed presenter yet**, the next
move is the call-for-presenters email (Step 7).

### Step 6 — commit and push (only when the chair approves)

The published branch is **`gh-pages`**. Commit the new
`ietf<NNN>/agenda.md` and push. Because CI lands its own "update indices"
commit on `gh-pages` after every push, a second push in the same session
often isn't a fast-forward — `git fetch` and `git rebase origin/gh-pages`
(your change only touches `agenda.md`; CI only touches `README.md`, so no
conflict), then push again.

Follow the repo's commit conventions (short imperative subject; previous

> agenda commits read like "Rough in 124 agenda"). Read `~/.claude/CODE.md`

Author: .claude/CODE.md is local to this instance of the repo, don't talk about it specifically.

Author: Fixed in 0de7201 — dropped the repo-local file reference; now just respect the chair's local git/gh environment rules.

first for the chair's git/gh environment rules, and end commit messages

> with the `Co-Authored-By: Claude ...` trailer per the harness.

Author: ```suggestion
first for the chair's git/gh environment rules, and end commit messages
with a `Co-Authored-By: Claude ...` trailer.
```

Author: Applied in 0de7201, folded in with removing the file reference above.

### Step 7 — draft the call-for-presenters email

Send the authors/editors of every draft that needs time **but has no
confirmed presenter** the standard reminder. It does not enumerate drafts —
it points at the agenda and asks the standard questions. Model it on the
chair's previous "HTTPbis at IETF<NNN> <City>" mails (search the chair's

> Sent mail for them). Current template:

Author: don't assume that the chair has sent them before. Just use the template below.

Author: Fixed in 0de7201 — Step 7 now just says to use the template below.
